packages/plugins/typescript/graphql-request/package.json

```diff
@@ -50,7 +50,8 @@
   "devDependencies": {
     "@graphql-codegen/testing": "1.18.3",
     "@graphql-tools/schema": "10.0.23",
-    "graphql-request": "6.0.0"
+    "graphql-request": "7.1.2",
+    "node-fetch": "2.7.0"
```

Comment:
`graphql-request@6` uses `cross-fetch`, whilst `graphql-request@7` uses native fetch (`undici`)
`undici` [is built in a way]((https://github.com/nodejs/undici/issues/531#issuecomment-815968075) ) that nock v14 and below does not support (support is coming in [nock v15](https://github.com/nock/nock/issues/2183#issuecomment-2888277507))

So, for tests to work, we need to pass a `fetch` function that `nock` supports, because `mockGraphQLServer` from `@graphql-codegen/testing` is using `nock` v13.4